controllers/postController.js

```diff
@@ -0,0 +1,131 @@
+const db = require('../models');
+
+module.exports = {
+  // Get all posts for homepage
+  allPosts: async (req, res, next) => {
+    try {
+      const results = await db.Post.find({}).sort({ date: -1 });
+      res.status(200).json(results);
+    } catch (err) {
+      next(err);
+    }
+  },
+  // Get a single post based on post ID
+  getPost: async (req, res, next) => {
+    try {
+      const result = await db.Post.findById(req.params.id);
+      res.status(200).json(result);
+    } catch (err) {
+      next(err);
+    }
+  },
+  // Get all posts by a specific user
+  getUserPosts: async (req, res, next) => {
+    try {
+      const results = await db.Post.find({ posterId: req.params.id }).sort({
+        date: -1
+      });
+      res.status(200).json(results);
+    } catch (err) {
+      next(err);
+    }
+  },
+  // Save a new post
+  savePost: async (req, res, next) => {
+    try {
+      const {
+        title,
+        summary,
+        description,
+        categories,
+        technologies,
+        posterId
+      } = req.body;
```

Comment:
Move down try block to start at first await
```suggestion
    const {
      title,
      summary,
      description,
      categories,
      technologies,
      posterId
    } = req.body;
    try {
```